source/common/filesystem/posix/filesystem_impl.cc

```diff
@@ -75,8 +75,23 @@ FileImplPosix::FlagsAndMode FileImplPosix::translateFlag(FlagSet in) {
   return {out, mode};
 }
 
+FilePtr InstanceImplPosix::createFile(const FilePathAndType& file_info) {
+  switch (file_info.file_type_) {
+  case DestinationType::File:
+    return std::make_unique<FileImplPosix>(file_info);
+  case DestinationType::Console:
+    return std::make_unique<ConsoleFileImplPosix>();
+  case DestinationType::Stderr:
+    return std::make_unique<StdErrFileImplPosix>();
+  case DestinationType::Stdout:
+    return std::make_unique<StdOutFileImplPosix>();
```

Comment:
Do we need separate types for each of these, or should we consider having the type -> path conversion inside the createFile method?

Reply:
It was to make it symmetrical in Posix and Windows but it is not needed so I removed these types on POSIX